csharp/ql/src/semmle/code/csharp/ir/implementation/raw/internal/TranslatedFunction.qll

```diff
@@ -0,0 +1,424 @@
+import csharp
+import semmle.code.csharp.ir.implementation.raw.IR
+private import semmle.code.csharp.ir.implementation.Opcode
+private import semmle.code.csharp.ir.internal.IRUtilities
+private import semmle.code.csharp.ir.implementation.internal.OperandTag
+private import semmle.code.csharp.ir.internal.TempVariableTag
+private import InstructionTag
+private import TranslatedElement
+private import TranslatedExpr
+private import TranslatedInitialization
+private import TranslatedStmt
+private import semmle.code.csharp.ir.internal.IRCSharpLanguage as Language
+
+/**
+ * Gets the `TranslatedFunction` that represents function `callable`.
+ */
+TranslatedFunction getTranslatedFunction(Callable callable) {
+  result.getAST() = callable
+}
+
+/**
+ * Represents the IR translation of a function. This is the root elements for
+ * all other elements associated with this function.
+ */
+class TranslatedFunction extends TranslatedElement,
+  TTranslatedFunction {
+  Callable callable;
+
+  TranslatedFunction() {
+    this = TTranslatedFunction(callable)
+  }
+
+  override final string toString() {
+    result = callable.toString()
+  }
+
+  override final Language::AST getAST() {
+    result = callable
+  }
+
+  /**
+   * Gets the function being translated.
+   */
+  override final Callable getFunction() {
+    result = callable
+  }
+
+  override final TranslatedElement getChild(int id) {
+    id = -2 and result = getConstructorInitializer() or
```

Comment:
Should I fix this in all the files for this PR?

Reply:
Ideally, but it also depends on what C++ QL is doing - it may be better to be consistent with that? Or maybe, we can persuade Dave to add `this.` everywhere.

Reply:
Since this is the internal raw implementation, the problem of keeping the style the same shouldn't be too pressing. I'll start fixing this.